lib/Net/Works/Address.pm

```diff
@@ -112,7 +112,7 @@ sub as_ipv4_string {
         'Cannot represent IP address larger than 2**32-1 as an IPv4 string'
         if $self->as_integer() >= 2**32;
 
-    return __PACKAGE__->new_from_integer(
+    return ( ref $self )->new_from_integer(
```

Comment:
Wouldn't `$self->new_from_integer` work just as well?

Reply:
I must admit I don't entirely understand the internals of Moo, but I do know it's terribly inconsistent when it comes to calling the `new` method on an existing object ( `new_from_integer` et al. chain to `new`.)

```
perl -e 'package Foo; use Moo; package main; my $foo = Foo->new->new'
Attempt to bless into a reference at /opt/markperl/lib/site_perl/5.28.1/Moo/Object.pm line 26.
```

But

```
perl -e 'package Foo; use Moo; has bar => is => "ro"; package main; my $foo = Foo->new->new'
(no error)
```

Now while Moo /currently/ does what we want when we call `new` on an existing Net::Works::Address instance, I'm not sure it will in the future, so I'm being explicit.